// react-native-flex-grid/lib/commonjs/components/Layout/Row/Row.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactNative = require("react-native");
var _grid = require("../../../utils/grid");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
const styles = _reactNative.StyleSheet.create({
  row: {
    flexDirection: 'row',
    flexWrap: 'wrap'
  },
  rtl: {
    flexDirection: 'row-reverse'
  }
});

/** Row */
const Row = _ref => {
  let {
    style,
    gx = 4,
    dir = 'ltr',
    Element = _reactNative.View,
    ...rest
  } = _ref;
  return /*#__PURE__*/_react.default.createElement(Element, _extends({
    style: [styles.row, dir === 'rtl' && styles.rtl, gx && {
      marginHorizontal: -1 * (0, _grid.getConfig)().gutters[gx] / 2
    }, style]
  }, rest));
};
var _default = Row;
exports.default = _default;
